src/components/WeatherList.js

```diff
@@ -0,0 +1,38 @@
+import React, { Component } from 'react';
+import WeatherForm from './WeatherForm';
+import Weather from './Weather';
+
+class WeatherList extends Component {
+    state = {
+
+        list: [
+            { conditions: "Rainy", highF: 62, lowF: 55 },
+            { conditions: "Cloudy", highF: 52, lowF: 35 },
+        ]
+    }
+
+
+    handleAddEntry = (entry) => {
+        this.setState(prev => {
+            const newList = prev.list.slice(0);
+            newList.push(entry);
+            return { list: newList };
+        })
+    }
+
+
+    render() {
+
+        return (
+            <div className="WeatherList">
+                <WeatherForm onAdd={this.handleAddEntry} foo="bar"/>
+                {this.state.list.map((data, index) =>
+                    <Weather key={index} conditions={data.conditions} highF={data.highF} lowF={data.lowF} />
```

Comment:
You want to base that `key` on more than just the index because React uses this key to identify if an element should be re-rendered. Because this is just the _position_ of the thing in the array, it's not the best indicator that it might need to be re-renders. More [here](https://reactjs.org/docs/lists-and-keys.html).